### PR TITLE
Differentiated configuration for RocksDB

### PR DESCRIPTION
**Pull request by qiujiayu**
### Motivation:

#903 

### Modification:

Describe the idea and modifications you've done.

### Result:

Fixes #<GitHub issue number>.

If there is no issue then describe the changes introduced by this PR.

### CODE REVIEW

**Comment by killme2008** [comment]
Good job. But looks like it makes a breaking API change. Can we still reserve the functions`registerXXXX(final Class<?> cls, final DBOptions opts)`?  Make them configure all raft groups' storage options by default.

**Comment by qiujiayu** [comment]
> Good job. But looks like it makes a breaking API change. Can we still reserve the functions`registerXXXX(final Class<?> cls, final XXXOptions opts)`? Make them configure all raft groups' storage options by default.

done

**Review by killme2008 — APPROVED** [review]
LGTM. But prefer to rename the property name.  Ping @fengjiachun

File: jraft-core/src/main/java/com/alipay/sofa/jraft/storage/impl/RocksDBLogStorage.java [hunk 1]
```diff
@@ -137,7 +137,10 @@ protected static class EmptyWriteContext implements WriteContext {
         static EmptyWriteContext INSTANCE = new EmptyWriteContext();
     }
 
+    public static final String PART_ROCKSDB_OPTIONS_KEY = "log.part.rocksdb.options";
```
[the remark this answers is not in the document]
Comment (fengjiachun) [reply jraft-core/src/main/java/com/alipay/sofa/jraft/storage/impl/RocksDBLogStorage.java:140]: I couldn't agree more, and all other configuration items follow this convention: `jraft.xxx`

Comment (fengjiachun) [inline jraft-core/src/main/java/com/alipay/sofa/jraft/storage/impl/RocksDBLogStorage.java:140]:
```suggestion
    public static final String PART_ROCKSDB_OPTIONS_KEY = "jraft.log.part.rocksdb.options";
```

**Pull request merged**